src/main/java/reposense/parsers/CliArgumentsParser.java

```diff
@@ -0,0 +1,96 @@
+package reposense.parsers;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Optional;
+
+import reposense.exceptions.ParseException;
+
+/**
+ * Verifies and parses user-supplied CLI arguments into an InputParameter Object.
+ */
+public class CliArgumentsParser extends Parser<InputParameter, String[]> {
+
+    private static final String ARGUMENT_PREFIX_DASH = "-";
+
+    private static final String CONFIG_FILE_ARG = "config";
+    private static final String TARGET_FILE_ARG = "output";
+    private static final String SINCE_DATE_ARG = "since";
+    private static final String UNTIL_DATE_ARG = "until";
+
+    private static final String PARSE_EXCEPTION_MESSAGE_INVALID_INPUTS = "Failed to parse inputs arguments";
+    private static final String PARSE_EXCEPTION_MESSAGE_NO_CSV_FILE = "Failed due to missing CSV file";
+
+    /**
+     * Parses user-supplied arguments into an InputParameter object.
+     *
+     * @throws ParseException If the given inputs or the csv file fail to parse or mandatory fields are missing.
+     */
+    @Override
+    public InputParameter parse(String[] args) throws ParseException {
```

Comment:
This can probably be turned into a static method as it doesn't use any instance variables